tools/ob_admin/backup_tool/ob_admin_backup_validation_task.cpp

```diff
@@ -0,0 +1,4508 @@
+        ret = OB_ERR_UNEXPECTED;
+        STORAGE_LOG(WARN, "archive scn is less than restore scn", K(ret), K(backup_piece_key),
+                    K(archive_scn), K(restore_scn));
+      }
+    } else if (backup_piece_attr->backup_piece_info_desc_->piece_.is_frozen()) {
+      raw_path_parent.set(dir_array, ls_attr->single_ls_info_desc_->checkpoint_scn_);
+      restore_scn = ls_attr->single_ls_info_desc_->checkpoint_scn_;
+      class GetSourceFunctor
+      {
+      public:
+        GetSourceFunctor(logservice::ObRemoteRawPathParent &raw_path_parent)
+            : raw_path_parent_(raw_path_parent)
+        {
+        }
+        int operator()(const share::ObLSID &id, logservice::ObRemoteSourceGuard &guard)
+        {
+          int ret = OB_SUCCESS;
+          logservice::ObRemoteRawPathParent *raw_path_parent
+              = static_cast<logservice::ObRemoteRawPathParent *>(
+                  logservice::ObResSrcAlloctor::alloc(share::ObLogRestoreSourceType::RAWPATH, id));
+          if (OB_ISNULL(raw_path_parent)) {
+            ret = OB_ALLOCATE_MEMORY_FAILED;
+            CLOG_LOG(WARN, "failed to allocate raw_path_parent", K(id));
+          } else if (OB_FAIL(raw_path_parent_.deep_copy_to(*raw_path_parent))) {
+            CLOG_LOG(WARN, "failed to deep copy to raw_path_parent");
+          } else if (OB_FAIL(guard.set_source(raw_path_parent))) {
+            CLOG_LOG(WARN, "failed to set location source");
+          }
+
+          if (OB_FAIL(ret) && OB_NOT_NULL(raw_path_parent)) {
+            logservice::ObResSrcAlloctor::free(raw_path_parent);
+            raw_path_parent = nullptr;
+          }
+          return ret;
+        }
+
+      private:
+        logservice::ObRemoteRawPathParent &raw_path_parent_;
+      };
+      GetSourceFunctor get_source_func(raw_path_parent);
+      logservice::ObRemoteLogGroupEntryIterator remote_iter(get_source_func);
+      logservice::LogGroupEntry tmp_entry;
+      const char *tmp_buf = nullptr;
+      int64_t tmp_buf_len = 0;
+
+      if (OB_FAIL(remote_iter.init(
+              backup_piece_attr->backup_piece_info_desc_->piece_.key_.tenant_id_, ls_id,
+              share::SCN(), palf::LSN(lsn_range.second.first), palf::LSN(lsn_range.second.second),
+              &piece_validation_dag->large_buffer_pool_, &piece_validation_dag->storage_handler_,
+              64L * 1024 * 1024 /*64M*/))) {
+        STORAGE_LOG(WARN, "failed to init remote_iter");
+      }
+      while (OB_SUCC(ret)) {
+        if (OB_FAIL(remote_iter.next(tmp_entry, tmp_lsn, tmp_buf, tmp_buf_len))) {
+          if (OB_ITER_END == ret) {
+            ret = OB_SUCCESS;
+            STORAGE_LOG(DEBUG, "succeed to iterate log", K(ret));
+            break;
+          } else {
+            STORAGE_LOG(WARN, "failed to iterate remote_log", K(ret), K(ls_id),
+                        K(palf::LSN(lsn_range.second.first)),
+                        K(palf::LSN(lsn_range.second.second)));
```

Comment:
should not use function in K(), consider. "xxx" : palf:LSN(lsn_range.second.first)

Reply:
fixed